corona-warn-app/cwa-app-android
# Contact Journal Event/Risk extension (EXPOSUREAPP-5413)

DRAFT-PR for Exposureapp-5413. (_Event Check-In UI: Contact Journal Event extension_)

As a CWA user I want to if a risk was caused by attending an event, so that I can warn others attending the same event but do not use the CWA.

### toDos: (done)
- [x] some layout/logic adjustments 
- [x]  check/adjust for latest techspec/r2.0 changes
- [x]  string adjustments
- [x]  adjustment of existing journal tests & new tests

### Note (update):
- This PR implements the logic and necessary UI for the risk-based-event for the contact journal. 
- For having a final event based risk, the PR #2660 has to merged

### How to test:
- For final testing, risk event PRs have to be merged (see above)
- For testing the UI in advance you can create an automatic screenshot  e.g. (Screenshot attached)
- check _ContactDiaryOverviewViewModelTest_, _ContactDiaryDatabaseTest_ & _ContactDiaryOverviewFragmentTest_  

### Screenshot:
![screen](https://user-images.githubusercontent.com/46747780/112990501-6e690480-9166-11eb-845e-76afd016e793.png)
